axum/src/extract/state.rs

```diff
@@ -295,6 +295,44 @@ use std::{
 ///
 /// In general however we recommend you implement `Clone` for all your state types to avoid
 /// potential type errors.
+///
+/// # When mutable shared state is needed
+/// As state is used across routes it is not mutable by default. To mutate the state
+/// you will need to use an `Arc<Mutex>` or similar.
+/// You may have to use `tokio::sync::Mutex` instead of `std::sync::Mutex`,
+/// since `std::sync::Mutex` is not safe for use across threads, but this comes
```

Comment:
`std::sync::Mutex` **is** safe to use across threads. If you don't have multiple threads, there's even no point in using it at all.

The issue is using it across `.await` points because that is extremely likely to cause an intermittend deadlock, and that is an issue even on single threaded runtimes, no only multithreaded ones.

Reply:
By "using across await points" what I mean is: Keeping it locked across await points, holding a `MutexGuard` while `.await`ing.

Reply:
@FSMaxB fixed in commit 856ac49.